### DISCUSSION TITLE
What do you use to query date

### DISCUSSION THREAD

**Discussion by bmichotte** [prisma/prisma#9809] [Q&A] [answered]
Hi there,

If I have the following model 
```
model Foo {
    id        String    @id @default(cuid())
    createdAt DateTime  @default(now())
}
```

I can, on my pgsql database, query all the `Foo` created on a certain date with 
```sql
select * from "Foo" where "createdAt"::date = '2021-10-17'
```
or with 
```sql
select * from "Foo" where CAST("createdAt" AS DATE) = '2021-10-17'
```

Is there any way to do it with prisma ?

Actually, the only way I found is 
```js
await prisma.foo.findMany({
   where: {
      AND: [
          { createdAt: { gte: new Date('2021-10-17 00:00:00') }},
          { createdAt: { lte: new Date('2021-10-17 23:59:59') }}
      ]
   }
})
```

which is not convenient and could lead to errors.

Is the only way to get that query is to create the model with another `DateTime @db.Date` field which will allow to use `{ theField: new Date() }` ?

**Comment by TasinIshmam** [comment]
Hey @bmichotte :wave:  

Could you clarify what you mean by the method you mentioned could lead to errors? 

Using `lt/lte` and `gt/gte` is the way to query date ranges in Prisma as of now.

**Reply by bmichotte** [reply]
Hello @TasinIshmam,

I think this method is more "dangerous" when we play with `lt(e)` and `gt(e)` as well as timezones than the "Date" version where you only check on a date (when you don't need to care of the time).

But anyway, thanks for the answer :D

**Reply by TasinIshmam** [reply]
Hey @bmichotte 

Regarding your concern about timezones: Prisma stores all dates in UTC and converts them appropriately before querying, so you should not face issues with timezones.

**Comment by pjtf93** [comment]
Hey @bmichotte did you find a solution for this?

**Reply by bmichotte** [reply]
Unfortunately no, I still use the `lte / gte` trick

**Reply by bmichotte** [reply]
Btw, I opened https://github.com/prisma/prisma/issues/11734

**Reply by jharrell** [reply, marked as answer]
Closing as issue #11734 was resolved. Thank you @bmichotte !
